exercises/practice/rational-numbers/.meta/example.f90

```diff
@@ -0,0 +1,95 @@
+
+module rational_numbers
+  implicit none
+contains
+
+  function add(r1,r2)
+    integer,dimension(2) :: add, r1,r2
+    !(a1 * b2 + a2 * b1) / (b1 * b2)
+    add(1) = r1(1)*r2(2)+r2(1)*r1(2)
+    add(2) = r1(2)*r2(2)
+    add = reduce(add)
+  end function
+
+  function sub(r1,r2)
+    integer,dimension(2) :: sub, r1,r2
+    !r₁ - r₂ = a₁/b₁ - a₂/b₂ = (a₁ * b₂ - a₂ * b₁) / (b₁ * b₂).
+    sub(1) = r1(1)*r2(2)-r2(1)*r1(2)
+    sub(2) = r1(2)*r2(2)
+    sub = reduce(sub)
+  end function
+
+  function mul(r1,r2)
+    integer,dimension(2) :: mul, r1,r2
+    !r₁ * r₂ = (a₁ * a₂) / (b₁ * b₂).
+    mul(1) = r1(1)*r2(1)
+    mul(2) = r1(2)*r2(2)
+    mul = reduce(mul)
+  end function
+
+  function div(r1,r2)
+    integer,dimension(2) :: div, r1,r2
+    ! r₁ / r₂ = (a₁ * b₂) / (a₂ * b₁)
+    div(1) = r1(1)*r2(2)
+    div(2) = r1(2)*r2(1)
+    div = reduce(div)
+  end function
+
+  function rabs(r1)
+    integer,dimension(2) :: rabs, r1
+    rabs(1) = abs(r1(1))
+    rabs(2) = abs(r1(2))
+  end function
+
+
+  function exprational(r1, ex)
```

Comment:
What do you think of longer function names like `rational_to_pow`?

Reply:
OK